src/skin.js

```diff
@@ -50,35 +53,39 @@ var onPlayerEvent = function(event, eventData) {
                 }
             }
 
+            this._controls.clickThroughLink.setUrl(clickThroughURL);
             this.setControls(linearAdStartedControls);
             this._controls.adCounter.update();
+            this._controls.videoStartCountdown.setCurrentAdDuration(getCreativeDuration(eventData.ad.getCoreAd()));
             break;
         case OO.Pulse.AdPlayer.Events.LINEAR_AD_FINISHED:
         case OO.Pulse.AdPlayer.Events.LINEAR_AD_SKIPPED:
-            this.setControls([ 'loadingSpinner' ]);
+            this.setControls([ 'loadingSpinner', 'videoStartCountdown']);
             this._controls.skipCountdown.setAd(null);
+            this._controls.videoStartCountdown.reduceAggregatedTime();
             break;
         case OO.Pulse.AdPlayer.Events.LINEAR_AD_PROGRESS:
-            this._controls.progressBar.setProgress(eventData.position / eventData.duration);
+            this._controls.progressBar.setProgress(eventData.duration, eventData.position);
+            this._controls.videoStartCountdown.updateTime(eventData.position);
             if(!this._isPlayingVPAID) {
                 this._controls.skipCountdown.update(eventData.position);
             }
             break;
         case OO.Pulse.AdPlayer.Events.LINEAR_AD_PAUSED:
             if(!this._isPlayingVPAID) {            
-                this.setControls([ 'muteButton', 'adCounter', 'progressBar', 'playButton' ]);
+                this.setControls([ 'muteButton', 'adCounter', 'progressBar', 'playButton', 'hoverOverlay', 'videoStartCountdown', 'clickThroughLink', 'controlContainer', 'fullScreenButton' ]);
```

Comment:
Maybe we can have something like this . 
```
var commonControls = ['muteButton', 'adCounter', 'progressBar', 'videoStartCountdown', 'fullScreenButton', 'controlContainer', 'clickThroughLink']
```
then in the place of setting controls we do
```
this.setControls(commonControls.concat(['hoverOverlay']))
```

Reply:
we don't have common controls across all events.